cmd/git/main_test.go

```diff
@@ -20,20 +23,45 @@ import (
 	shpgit "github.com/shipwright-io/build/pkg/git"
 )
 
-var _ = Describe("Git Resource", func() {
-	var run = func(args ...string) error {
-		// discard log output
-		log.SetOutput(io.Discard)
+type opts struct {
+	ctx       context.Context
+	logOutput io.Writer
+	args      []string
+}
+
+type runOpts func(*opts)
+
+func withLogOutput(out io.Writer) runOpts { return func(o *opts) { o.logOutput = out } }
+func withArgs(args ...string) runOpts     { return func(o *opts) { o.args = args } }
+
+func run(o ...runOpts) error {
```

Comment:
Moved the `run` function from within the `Describe` into the global context and changed it that it is more flexible.